fine_tuning_job.go

```diff
@@ -23,8 +24,40 @@ type FineTuningJob struct {
 	TrainedTokens   int             `json:"trained_tokens"`
 }
 
+type HyperparameterNEpochs struct {
+	IntValue    *int    `json:"-"`
+	StringValue *string `json:"-"`
+}
+
+func (h *HyperparameterNEpochs) UnmarshalJSON(data []byte) error {
+	var intValue int
+	var stringValue string
+
+	if err := json.Unmarshal(data, &intValue); err == nil {
+		h.IntValue = &intValue
+		return nil
+	}
+
+	if err := json.Unmarshal(data, &stringValue); err != nil {
+		return err
+	}
+
+	h.StringValue = &stringValue
+	return nil
+}
+
+func (h *HyperparameterNEpochs) MarshalJSON() ([]byte, error) {
+	if h.IntValue != nil {
+		return json.Marshal(*h.IntValue)
+	} else if h.StringValue != nil {
+		return json.Marshal(*h.StringValue)
+	}
+
+	return nil, fmt.Errorf("invalid hyperparameter n_epochs")
+}
+
 type Hyperparameters struct {
-	Epochs int `json:"n_epochs"`
+	Epochs *HyperparameterNEpochs `json:"n_epochs,omitempty"`
```

Comment:
Hey, thank you so much for the PR! I wonder if we can use `any` here to avoid `Epochs: &HyperparameterNEpochs{` initialization

Reply:
done